arm-documentdb/2015-04-08/documentdb.json

```diff
@@ -0,0 +1,767 @@
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DatabaseAccountRegenerateKeyParameters"
+            },
+            "description": "Specifies name of the key which should be regenerated."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "",
+            "schema": {
+              "$ref": "#/definitions/DatabaseAccountListKeysResult"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.DocumentDB/databaseAccountNames/{accountName}?api-version={api-version}": {
+      "head": {
+        "operationId": "DatabaseAccounts_CheckNameAvailability",
+        "description": "Checks that account name is valid and is not in use.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/accountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The account name is valid but already unavailable (already in use)."
+          },
+          "404": {
+            "description": "Not Found. The account name is available and valid."
+          },
+          "400": {
+            "description": "Bad Request. The account name is invalid."
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "DatabaseAccountsListResult": {
+      "properties": {
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DatabaseAccount"
+          },
+          "description": "List of database account and their properties."
+        }
+      },
+      "description": "The list database accounts operation response."
+    },
+    "FailoverPolicies": {
+      "type": "array",
+      "description": "List of failover policies.",
+      "items": {
+        "$ref": "#/definitions/FailoverPolicy"
+      }
+    },
+    "FailoverPolicy": {
+      "type": "object",
+      "description": "The failover policy for a given region of a database account.",
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The unique identifier of the region within the database account. Example: <accountName>-<locationName>"
+        },
+        "locationName": {
+          "type": "string",
+          "description": "The name of the region in which the database account exists."
+        },
+        "failoverPriority": {
+          "type": "integer",
+          "description": "The failover priority of the region. A failover priority of 0 indicates a write region. The maximum value for a failover priority = (total number of regions - 1). Failover priority values must be unique for each of the regions in which the database account exists."
```

Comment:
may be a constraint of `"minimum": 0`, if the service does not accept negative values.